### PR TITLE
Swap 1165 filter proposals by questionary questions

### PR DESCRIPTION
**Pull request by jekabs-karklins**
## Description

This PR aims to implement a use case where the user officer wants to find proposals based on answers provided in the questionary. 

## Motivation and Context

This functionality allows user officers to filter proposals based on filer criteria. The filtering has been so far implemented for 4 question types. Text, Boolean, Date, and Multiple choice.
image

## Fixes

SWAP-1165

## Changes

Proposals page

## Depends on
https://github.com/UserOfficeProject/user-office-backend/pull/170

## Tests included/Docs Updated?

- [x] I have added tests to cover my changes.
- [ ] All relevant doc has been updated

### CODE REVIEW

**Comment by bolmsten** [comment]
When changing the question one search for the answer is pre-populated with the old

**Comment by bolmsten** [comment]
It is possible to fill in the date format wrong and get an error

**Comment by bolmsten** [comment]
The dropdown filter list is missing options as it only takes the "original" values declared

File: cypress/integration/proposal_administration.ts [hunk 1]
```diff
@@ -178,4 +178,216 @@ context('Proposal administration tests', () => {
...
+      .invoke('html')
+      .then(fieldId => {
+        textQuestionId = fieldId;
+      });
+  });
+
+  const proposalTitle = 'visi te tusee'; // faker.random.words(3);
+  const answerDate = '2030-01-01';
+  const answerMultipleChoice = 'One';
+  const answerText = 'kamert vins klusee'; // faker.random.words(3);
```
Comment (peter-asztalos) [inline cypress/integration/proposal_administration.ts:258]: Not sure, but maybe you forgot to use the faker version after testing?

**Review by peter-asztalos — APPROVED** [review]
Looks good (☞ﾟヮﾟ)☞

File: cypress/integration/proposal_administration.ts (hunk 1 above)
Comment (jekabs-karklins) [reply cypress/integration/proposal_administration.ts]: Thanks

**Review by RasmiaKulan — APPROVED** [review]

**Pull request merged**